falcon/request_helpers.py

```diff
@@ -129,3 +134,271 @@ def readlines(self, hint=None):
+        """Return the value of a query string parameter as a list.
+
+        List items must be comma-separated or must be provided
+        as multiple instances of the same param in the query string
+        ala *application/x-www-form-urlencoded*.
+
+        Args:
+            name (str): Parameter name, case-sensitive (e.g., 'ids').
+            transform (callable, optional): An optional transform function
+                that takes as input each element in the list as a ``str`` and
+                outputs a transformed element for inclusion in the list that
+                will be returned. For example, passing ``int`` will
+                transform list items into numbers.
+            required (bool, optional): Set to ``True`` to raise
+                ``HTTPBadRequest`` instead of returning ``None`` when the
+                parameter is not found (default ``False``).
+            store (dict, optional): A ``dict``-like object in which to place
+                the value of the param, but only if the param is found (default
+                ``None``).
+
+        Returns:
+            list: The value of the param if it is found. Otherwise, returns
+            ``None`` unless required is True. Empty list elements will be
+            discarded. For example a query string containing this::
+
+                things=1,,3
+
+            or a query string containing this::
+
+                things=1&things=&things=3
+
+            would both result in::
+
+                ['1', '3']
+
+        Raises:
+            HTTPBadRequest: A required param is missing from the request.
+            HTTPInvalidParam: A tranform function raised an instance of
+                ``ValueError``.
+
+        """
+
+        params = self.source
+
+        # PERF: Use if..in since it is a good all-around performer; we don't
+        #       know how likely params are to be specified by clients.
+        if name in params:
+            items = params[name]
+
+            # NOTE(warsaw): When a key appears multiple times in the request
+            # query, it will already be represented internally as a list.
+            # NOTE(kgriffs): Likewise for comma-delimited values.
+            if not isinstance(items, list):
+                items = [items]
+
+            # PERF(kgriffs): Use if-else rather than a DRY approach
+            # that sets transform to a passthrough function; avoids
+            # function calling overhead.
+            if transform is not None:
+                try:
+                    items = [transform(i) for i in items]
+
+                except ValueError:
+                    msg = 'The value is not formatted correctly.'
+                    raise HTTPInvalidParam(msg, name)
+
+            if store is not None:
+                store[name] = items
+
+            return items
+
+        if not required:
+            return None
+
+        raise HTTPMissingParam(name)
```

Comment:
I think this makes sense; the only downside is that magic methods a slightly slower than the alternative. As long as we mention this in the docs, should be OK.